alkos/kernel/arch/x86_64/abi/debug/debug_terminal.cpp

```diff
@@ -0,0 +1,20 @@
+/* internal includes */
+#include <debug_terminal.hpp>
+#include <kernel_assert.hpp>
+#include <serial_port_qemu/serial_qemu.hpp>
+
+void DebugTerminalWriteArch_(const char *const buffer)
```

Comment:
The arch should just provide the DebugTerminalWrite
i dont  see any benefit in making a wrapper?
Matter of fact I think there should be a .h file called Abi.hpp with all the functions that the arch provides.
This would solve the issue i think

Reply:
In essence i believe this file shoudn't exists unless there is some very specific benefit or use case in making a wrapper

Reply:
This was a try to not use ifdef macros but instead constexpr ifs, otherwise I would use ifdef as usual. At this stage this is only a test to present example solution to the team. To be rised on the sync